atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/ValidatingTransactionScopedCache.java

```diff
@@ -140,6 +180,21 @@ private void validateCacheReads(
         }
     }
 
+    private void validateCacheRowReads(
+            TableReference tableReference,
+            NavigableMap<byte[], RowResult<byte[]>> remoteReads,
+            NavigableMap<byte[], RowResult<byte[]>> cacheReads) {
+        if (!ByteArrayUtilities.areRowResultsEqual(remoteReads, cacheReads)) {
+            log.error(
+                    "Reading from lock watch cache returned a different result to a remote read - this indicates there "
+                            + "is a corruption bug in the caching logic",
+                    UnsafeArg.of("table", tableReference));
+            failureCallback.run();
+            throw new TransactionLockWatchFailedException(
```

Comment:
nit: The last 2 lines are same as `validateCacheReads` so we could extract this out.